massa-protocol-worker/src/handlers/block_handler/retrieval.rs

```diff
@@ -734,17 +734,24 @@ impl RetrievalThread {
         );
 
         // check if we were looking to retrieve the list of ops for that block
-        let wishlist_info = if let Some(info) = self.block_wishlist.get_mut(&block_id) && info.header.is_some() && info.operation_ids.is_none() {
-            // we were actively looking for this data
-            info
+        let wishlist_info = if let Some(info) = self.block_wishlist.get_mut(&block_id) {
```

Comment:
cleaner approach to avoid duplication:

```rust

let wishlist_info = match self.block_wishlist.get_mut(&block_id).filter(|info| info.header.is_some() && info.operation_ids.is_none()) {
    Some(v) => v,
    None => {
        ...
        return;
    }
}



```

Reply:
Done